### PR TITLE
Python3 support for restapi

### PR DESCRIPTION
**Pull request by waychal**

### CODE REVIEW

File: aiida/restapi/resources.py [hunk 1]
```diff
@@ -165,10 +169,11 @@ def get(self, id=None, page=None):
             url=url,
             url_root=url_root,
             path=request.path,
-            id=id,
-            query_string=request.query_string,
+            id=node_id,
+            query_string=query_string,
```
[the remark this answers is not in the document]
Comment (waychal) [reply aiida/restapi/resources.py]: Unquoted and quoted query string in response object both works in same way.

File: aiida/restapi/common/utils.py [hunk 2]
```diff
@@ -442,18 +446,19 @@ def build_response(self, status=200, headers=None, data=None):
         response.status_code = status
 
         if headers is not None:
-            for k, v in headers.items():
-                response.headers[k] = v
+            for key, val in headers.items():
+                response.headers[key] = val
```
Comment (dev-zero) [inline aiida/restapi/common/utils.py]: for next time, possibly use `respone.headers.update(headers)` as a shorthand

File: aiida/restapi/translator/data/structure.py [hunk 3]
```diff
@@ -41,31 +42,29 @@ def __init__(self, **kwargs):
...
         Create the basic query_help
         """
-        super(StructureDataTranslator, self).__init__(Class=self.__class__,
-                                                      **kwargs)
-
+        super(StructureDataTranslator, self).__init__(Class=self.__class__, **kwargs)
 
     @staticmethod
-    def get_visualization_data(node, format=None):
+    def get_visualization_data(node, visformat=None):
```
Comment (dev-zero) [inline aiida/restapi/translator/data/structure.py]: why not put `visformat="xsf"` here directly? then its clear by looking at the signature of the function what the default is

File: aiida/restapi/translator/data/structure.py [hunk 4]
```diff
@@ -74,30 +73,28 @@ def get_visualization_data(node, format=None):
 
         return response
 
-
     @staticmethod
-    def get_downloadable_data(node, format=None):
+    def get_downloadable_data(node, download_format=None):
```
Comment (dev-zero) [inline aiida/restapi/translator/data/structure.py]: ... same here

File: aiida/restapi/common/utils.py (hunk 2 above)
Comment (waychal) [reply aiida/restapi/common/utils.py]: Here response.header is not a type of dict so we can not directly update it with another dictionary.

**Comment by giovannipizzi** [comment]
This fixes #2086

**Pull request merged**